spec/support/server.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require "sinatra/base"
+
+class DiagnoseReport
+  attr_reader :params
+
+  def initialize(params: {}, report: {})
+    @params = params
+    @report = report
+  end
+
+  def section(key)
+    @report.fetch(key.to_s) { raise "No `#{key}` key found in the DiagnoseReport" }
+  end
+
+  def to_h
+    @report
+  end
+end
+
+class DiagnoseServer < Sinatra::Base
+  class << self
+    def run!(port)
+      @mutex = Mutex.new
+      @response_code = 200
+      @received_requests = []
+      super(:port => port)
+    end
+
+    def track_request(request)
+      @mutex.synchronize do
+        @received_requests << request
+      end
+    end
+
+    def last_received_report
+      @mutex.synchronize do
+        request = @received_requests.last
+        if request
+          DiagnoseReport.new(
+            :params => request.params,
+            :report => JSON.parse(request.body.read)
+          )
+        end
+      end
+    end
+
+    def clear!
+      @mutex.synchronize do
+        @response_code = nil
+        @received_requests.clear
+      end
+    end
+
+    def response_code
+      @mutex.synchronize do
+        @response_code
+      end
+    end
+
+    def response_code=(response_code)
+      @mutex.synchronize do
+        @response_code = response_code
+      end
+    end
+  end
+
+  set :logging, false
```

Comment:
[The internet says](https://github.com/sinatra/sinatra/pull/1153) that adding `set :quiet, false` will get rid of some of the extra logging messages on standard output (the cutesy ones regarding entering and exiting stages, I think).

Also, since this is a subclass of `Sinatra::Base`, [the docs say](http://sinatrarb.com/configuration.html) (ctrl-f ":logging") that it should already have `set :logging, false` by default. 

Additionally, per-request logging may be coming from the Rack logger itself, and [the docs say](http://sinatrarb.com/intro.html) (ctrl-f "logging") that you need to `set :logging, nil` (which for some reason is different than `set :logging, false`) to disable it entirely.

Reply:
Thanks!

Reply:
This seems to do the trick for most of it:

```ruby
  set :logging, false
  set :quiet, true
```

It will still print the output about the server starting though.

Reply:
I tested the things you suggested, but I can't get it to be quiet or disable logging with these options either 🤷‍♂️